pkg/controller/helper/tenant.go

```diff
@@ -0,0 +1,41 @@
+package helper
+
+import (
+	capabilitiesv1alpha1 "github.com/3scale/3scale-operator/apis/capabilities/v1alpha1"
+	porta_client_pkg "github.com/3scale/3scale-porta-go-client/client"
+)
+
+/*
+FetchTenant fetches tenant from 3scale
+- tenant
+- portaClient
+*/
+func FetchTenant(tenant *capabilitiesv1alpha1.Tenant, portaClient *porta_client_pkg.ThreeScaleClient) (*porta_client_pkg.Tenant, error) {
+	if tenant.Status.TenantId == 0 {
+		// tenantId not in status field
+		// Tenant has to be created
+		return nil, nil
+	}
+
+	tenantDef, err := portaClient.ShowTenant(tenant.Status.TenantId)
+	if err != nil && porta_client_pkg.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+	return tenantDef, nil
+}
+
+/*
+DeleteTenant deletes tenant from 3scale
+- tenant
+- portaClient
+*/
+func DeleteTenant(tenant *capabilitiesv1alpha1.Tenant, portaClient *porta_client_pkg.ThreeScaleClient) error {
```

Comment:
nitpick: this method does not add anything, does it?

Reply:
Potentially it could be moved, although I was thinking it would be nice to have it in a single file, I was thinking of possibility moving the tenant create here too, but happy to remove it directly if that's what you preffer.

Reply:
Updated